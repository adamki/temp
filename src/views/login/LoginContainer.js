import React from 'react';
import PropTypes from 'prop-types';
import logo from '../../logo.svg';
import LoginForm from './login';

import './styles.css';

class LoginContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {value: ''};
  }

  handleChange = event => {
    this.setState({value: event.target.value});
  }

  handleSubmit = event => {
    event.preventDefault();
    const address = this.state.value;

    this.props.history.push(`/addresses/${address}`);
  }

  render() {
   return (
     <div className="login">
       <section className="login__form">
         <img src={logo} className="login__logo" alt="company-logo" />
         <LoginForm
           value={this.state.value}
           handleSubmit={this.handleSubmit}
           handleChange={this.handleChange}
         />
       </section>
     </div>
    );
  }
}

LoginContainer.propTypes = {
  history: PropTypes.object.isRequired,
};

export default LoginContainer;
