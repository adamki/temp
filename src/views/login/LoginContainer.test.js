import React from 'react';
import ReactDOM from 'react-dom';
import axios from 'axios';

import renderer from 'react-test-renderer';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

import LoginContainer from './LoginContainer';

describe('Container: LoginContainer', () => {
  it('renders without crashing', () => {
    const container = renderer.create(<LoginContainer />);

    let tree = container.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an address input', () => {
    expect(
      mount(<LoginContainer />).find({'data-test': 'login-addressee'}).length,
    ).toEqual(1);
  });

  it('should pass a value to the onChange function', () => {
    const container = renderer.create(<LoginContainer />);
    let tree = container.toJSON();
    const component = mount(<LoginContainer />);

    expect(component.state().value).toBe('');

    component.find({'data-test': 'login-addressee'}).simulate('change', {
      target: {
        value: 'Jane Doe',
      },
    });

    expect(component.state().value).toBe('Jane Doe');
  });
});
