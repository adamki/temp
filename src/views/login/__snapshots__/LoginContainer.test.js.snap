// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: LoginContainer renders without crashing 1`] = `
<div
  className="login"
>
  <section
    className="login__form"
  >
    <img
      alt="company-logo"
      className="login__logo"
      src="logo.svg"
    />
    <article
      className="card"
      data-test="login-card"
    >
      <header
        className="card__header"
      >
        Welcome! Sign in With your Jobcoin Address
      </header>
      <section
        className="card__body"
      >
        <form
          onSubmit={[Function]}
        >
          <label>
            <h5>
              Jobcoin Address:
            </h5>
          </label>
          <input
            data-test="login-addressee"
            onChange={[Function]}
            type="text"
            value=""
          />
          <input
            type="submit"
            value="Submit"
          />
        </form>
      </section>
    </article>
  </section>
</div>
`;
